### PR TITLE
docs: document special host-gateway value for add-host

### PR DESCRIPTION
**Pull request by dvdksn**
Signed-off-by: David Karlsson <35727626+dvdksn@users.noreply.github.com>

**- What I did**

Add some docs on `--add-host host.docker.internal:host-gateway` and `dockerd --host-gateway-ip`

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

PXL_20230617_072054134

### CODE REVIEW

**Comment by dvdksn** [comment]
@thaJeztah does this LGTY?

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -759,24 +759,27 @@ PING docker (93.184.216.34): 56 data bytes
...
+connect to services running on the host machine.
 
-The flags you pass to `ip addr show` depend on whether you are
-using IPv4 or IPv6 networking in your containers. Use the following
-flags for IPv4 address retrieval for a network device named `eth0`:
+It's conventional to use `host.docker.internal` as the hostname referring to
+`host-gateway`. Docker Desktop automatically resolves this hostname, see
+[Explore networking features](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host).
+
+The following example creates a simple HTTP server that serves a file from the
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:770]: perhaps avoid `simple` :)

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -759,24 +759,27 @@ PING docker (93.184.216.34): 56 data bytes
...
+host to a container.
 
 ```console
-$ HOSTIP=`ip -4 addr show scope global dev eth0 | grep inet | awk '{print $2}' | cut -d / -f 1 | sed -n 1p`
-$ docker run  --add-host=docker:${HOSTIP} --rm -it debian
+$ echo "hello from host!" > ./hello
+$ python3 -m http.server 8000
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:775]: Was looking if we could avoid `python`, and avoid having to create a file, but looks like `nc` is _just_ different enough between mac and linux for it to not work great as example (Linux requires the `-N` option, and macOS doesn't, and doesn't support it).

```bash
echo -ne "HTTP/1.0 200 OK\r\n\r\nhello from host\r\n" | nc -N -l 8080
```


Do you know if python allows for the content to be read from stdin?

Comment (dvdksn) [reply docs/reference/commandline/run.md:775]: I don't think this python module can read from stdin - only a directory. Would also prefer a better option... But I couldn't come up with one :D Happy to take suggestions..

Comment (thaJeztah) [reply docs/reference/commandline/run.md:775]: Yeah, don't think there's a better one. At most we could just `grep` the `/etc/hosts` file inside the container to "show" that it's there, but showing what it's used for is nicer.

We could add some wording "to illustrate" or something (as in; don't need to start installing `python3` if you don't have it installed... just to see "it works").

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**